### DISCUSSION TITLE
Which maven scope is used by trivy ?

### DISCUSSION THREAD

**Discussion by sbernard31** [aquasecurity/trivy#5874] [Q&A] [answered]
### Question

**Trivy** is able to extract maven dependencies from `pom.xml`.

But which scope is used ? and it is configurable ? 

I see some information about that for version `0.22.0` (https://github.com/aquasecurity/trivy/discussions/1518) : 
> Dependencies whose scope is "test", "provided", “system” and "runtime" will be skipped.

But don't know if this is still true now. Maybe some new feature/enhancement was added ?

**Comment by knqyf263** [comment, marked as answer]
>But don't know if this is still true now. Maybe some new feature/enhancement was added ?

It should be true.

**Reply by sbernard31** [reply]
Thx :pray: !

So I guess if I want more tweaking about that, I should rather generate SBOM with something like [cyclonedx-maven-plugin](https://github.com/CycloneDX/cyclonedx-maven-plugin) then use **trivy** to scan vulnerabilites from that generated SBOM.

**Reply by knqyf263** [reply]
I think we can make it configurable if many users want that feature.

**Reply by sbernard31** [reply]
In my case, where I need to scan a code repository of the library I'm developing. 
I think that finally this is not so bad to use a full feature dedicated tools for my ecosystems/languages to generate SBOM and then use another specialize tools which consume SBOM to check vulnerabilities. (like trivy)

Concerning trivy, I don't know exactly what is its scope, so don't know if this really makes sense to add a lot of language specific feature. :shrug:
